.github/workflows/verify.yml

```diff
@@ -0,0 +1,35 @@
+name: Verify
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  actions: none
+  checks: none
+  contents: none
+  deployments: none
+  id-token: none
+  issues: none
+  discussions: none
+  packages: none
+  pages: none
+  pull-requests: none
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+on: [push, pull_request]
+
+jobs:
+  mingw:
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+    name: Meterpreter C Mingw Docker Build
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Compile
+        run: |
+          cd c/meterpreter
+          script --return --command 'make docker'
```

Comment:
The docker command runs with `-it` which requires a pty, so we have to wrap it in `script`